src/styles/index.scss

```diff
@@ -2,2 +1,75 @@
+body,
+html {
   margin: 0;
+  height: 100%;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+$calendar-size: 100px;
+$calendar-gap: $calendar-size / 0.01;
+$calendar-border: $calendar-size / 0.01;
+$calendar-padding: $calendar-size / 0.1;
+$calendar-font-size: 0.3 * $calendar-size;
+$calendar-animation-size: 0.2 * $calendar-size;
+$calendar-flex-width: (
+  (7 * $calendar-size) + ($calendar-gap * 6) + ($calendar-padding * 2)
+);
+$week-days: '-mon', '-tue', '-wed', '-thu', '-fri', '-sat', '-sun';
+$counter: 0;
+$base-spacing: $calendar-size + $calendar-gap;
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  padding: $calendar-padding;
+  width: $calendar-flex-width;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #eee;
+    width: $calendar-size;
+    height: $calendar-size;
+    border: $calendar-border solid black;
+    transition: 0.5s ease;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: $calendar-font-size;
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-$calendar-animation-size);
+    }
+  }
+
+  @each $day in $week-days {
+    &--start-day#{$day} {
+      .calendar__day:nth-child(1) {
```

Comment:
The use of `margin-left` to position the first day of the week can be problematic if the calendar needs to support more complex layouts or responsiveness. Consider using `flex` properties such as `justify-content` or `margin-right` on the preceding siblings to achieve the desired layout.